frontend/src/data/Badge.ts

```diff
@@ -1,36 +1,39 @@
-import URI from "urijs";
-
 import { Implementation } from "../data/parseReportData";
 import Dialect from "./Dialect";
 import siteURI from "./Site";
 
-const SHIELDS = new URI("https://img.shields.io/endpoint");
+const SHIELDS = new URL("https://img.shields.io/endpoint");
 
-const BADGES = siteURI.clone().segment("badges");
+const BADGES = new URL(siteURI.toString());
+BADGES.pathname += "/badges";
 
-const badgeFor = (uri: URI): URI => SHIELDS.clone().addQuery("url", uri);
+const badgeFor = (url: URL): URL =>
+  new URL(SHIELDS.toString() + "?url=" + encodeURIComponent(url.toString()));
```

Comment:
This also looks wrong. The whole point of using URL types is to completely avoid string manipulation, which is easily going to be wrong.

Reply:
Thanks! I just fixed this on my local branch.